src/App.tsx

```diff
@@ -1,51 +1,105 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
+import cn from 'classnames/bind';
 import './App.scss';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
+interface Good {
+  id: number;
+  name: string;
+}
+
+const SORT_BY_ABC = 'abc';
+const SORT_BY_LENGTH = 'length';
+
+function getReorderedGoods(
+  goods: Good[],
+  sortType: string,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SORT_BY_LENGTH) {
+    visibleGoods.sort((good1, good2) => good1.name.length - good2.name.length);
+  }
+
+  if (sortType === SORT_BY_ABC) {
+    visibleGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortField, reversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_BY_ABC,
+          })}
+          onClick={() => setSortField(SORT_BY_ABC)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
```

Comment:
Ensure that the active button is visually distinct to indicate the current sorting method, as per the checklist requirement.